File: gh-2777/test-server/src/main/scala/org/github/jleskovar/test/TestServerApplication.scala
package org.github.jleskovar.test

import org.springframework.boot.SpringApplication
import org.springframework.boot.actuate.metrics.repository.MetricRepository
import org.springframework.boot.actuate.metrics.repository.redis.RedisMetricRepository
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.context.annotation.Bean
import org.springframework.data.redis.connection.RedisConnectionFactory

@SpringBootApplication
class TestServerApplication {

  @Bean
  def metricRepository(redisConnectionFactory: RedisConnectionFactory): MetricRepository =
    new RedisMetricRepository(redisConnectionFactory)

}

object TestServerApplication extends App {
  SpringApplication.run(classOf[TestServerApplication])
}

File: gh-2777/test-server/src/main/java/org/github/jleskovar/test/TestServerApplication.java
package org.github.jleskovar.test;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.actuate.metrics.repository.MetricRepository;
import org.springframework.boot.actuate.metrics.repository.redis.RedisMetricRepository;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;
import org.springframework.data.redis.connection.RedisConnectionFactory;

@SpringBootApplication
public class TestServerApplication {

	@Bean
	public MetricRepository metricRepository(RedisConnectionFactory redisConnectionFactory) {
		return new RedisMetricRepository(redisConnectionFactory);
	}

	public static void main(String[] args) {
		SpringApplication.run(TestServerApplication.class);
	}

}
